### PR TITLE
misc: Update MAINTAINERS.yaml

### PR DESCRIPTION
**Pull request by BobbyRBruce**
- Updates the list of current gem5 Maintainers.
- Updates the MAINATAINERS.yaml comments to reflect gem5's contribution policy and maintainer responsibilities since moving to GitHub.
- Updates the subsystem maintainers, based on maintainer's stated preferences.
- Adds the optional 'expert' field to subsystems.

### CODE REVIEW

File: MAINTAINERS.yaml
Comment (powerjg) [inline MAINTAINERS.yaml]: Let's drop the PMC here. I would rather have this list outside the main repo. Maybe point to the maintainers team so we have a little more agility with this list. https://github.com/orgs/gem5/teams/maintainers

File: MAINTAINERS.yaml
Comment (powerjg) [inline MAINTAINERS.yaml:69]: @shingarov, do you want to continue to maintain this code?

File: MAINTAINERS.yaml
Comment (powerjg) [inline MAINTAINERS.yaml]: @abmerop, does it make sense to drop this tag? Do you want to continue to be the maintainer?

File: MAINTAINERS.yaml
Comment (powerjg) [inline MAINTAINERS.yaml:265]:
```suggestion
  status: orphaned
  experts:
    - Jason Lowe-Power <jason@lowepower.com>
```

**Comment by powerjg** [comment]
Generally, I would advocate for removing this file. I think it's a bit embarrassing how much of gem5 is "orphaned" according to this. It's also not really indicative of the current state. Almost anyone submitting improvements to these subsystems will get reviews. Though, the other way to look at it is that if anyone opened an issue in the "orphaned" components, it's likely that no one would be available to answer their question or fix their bug. So, in that sense, I guess they are orphaned.

Anyway, just some thoughts. No action needed.

**Review by abmerop — COMMENTED** [review]
If we are keeping this file, you can leave me as the maintainer for dev-hsa, dev-amdgpu, and gpu-compute.

Not sure if experts are meant to be mutually exclusive with maintainers, but you could add me as expert for dev-amdgpu if no.

**Comment by powerjg** [comment]
My interpretation is that maintainer implies expert.

File: MAINTAINERS.yaml
Comment (BobbyRBruce) [reply MAINTAINERS.yaml]: I'm not sure if you can see the team https://github.com/orgs/gem5/teams/maintainers unless you're a member of the gem5 organization. At least when i logout of GitHub and try to go there I don't see anything. I couldn't find any way to make it public. Given that I've opted just to remove this.

**Comment by BobbyRBruce** [comment]
@powerjg : I'm ok to merge this if you're comfortable giving it an approving review.

**Review by powerjg — APPROVED** [review]

File: MAINTAINERS.yaml
Comment (abmerop) [inline MAINTAINERS.yaml:63]: I know this is already merged and this isn't super important but my name is spelled incorrectly here including the email

Comment (powerjg) [reply MAINTAINERS.yaml:63]: Oops!! @ivanaamit can you fix this? It should be "Poremba" (unless I've again made an embarrassing error)

**Pull request merged**
